### PR TITLE
KAS: try to type some things

### PR DESCRIPTION
**Pull request by handeyeco**
## Summary:
Functionality shouldn't change, just trying to modernize some things.

### CODE REVIEW

File: packages/kas/package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
         "dist"
     ],
     "scripts": {
-        "gen:parsers": "node src/parser-generator.js",
+        "gen:parsers": "node src/parser-generator.ts",
```
Comment (jeremywiebe) [inline packages/kas/package.json:25]: Does this work (passing a TypeScript file to node directly)?

Previously we had to use SWC to run TS scripts as command line apps: https://github.com/Khan/perseus/blob/main/utils/pre-publish-check-ci.ts#L1

File: packages/kas/src/compare.ts [hunk 2]
```diff
@@ -1,12 +1,15 @@
-/* eslint-disable */
-import _ from "underscore";
```
Comment (jeremywiebe) [inline packages/kas/src/compare.ts:2]: Yay!

File: packages/kas/src/compare.ts [hunk 3]
```diff
@@ -16,18 +19,17 @@ export const compare = function (expr1, expr2, options) {
      */
 
     if (options !== undefined) {
-        // eslint-disable-next-line no-undef
-        options = _.extend(defaults, options);
+        options = {...defaults, ...options};
     } else {
         options = defaults;
     }
```
Comment (jeremywiebe) [inline packages/kas/src/compare.ts:25]: FWIW, with spread syntax, we can just spread the `options` (even if they're `undefined`).

image

**Review by mark-fitzgerald — APPROVED** [review]
Thanks for the clean-up!

File: packages/kas/package.json (hunk 1 above)
Comment (handeyeco) [reply packages/kas/package.json:25]: It seems to work when I run it 🤷

**Pull request merged**